apps/aecore/lib/aecore/multi_node_test_framework/worker.ex

```diff
@@ -0,0 +1,611 @@
+defmodule Aecore.MultiNodeTestFramework.Worker do
+  @moduledoc """
+  Module for multi node sync test
+  """
+
+  alias Aeutil.Serialization
+
+  require Logger
+  use GenServer
+
+  def start_link(_args) do
+    GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
+  end
+
+  def init(args) do
+    {:ok, args}
+  end
+
+  def get_state do
+    GenServer.call(__MODULE__, {:get_state})
+  end
+
+  @spec new_node(String.t(), non_neg_integer(), non_neg_integer()) :: :already_exists | Map.t()
+  def new_node(node_name, port, sync_port) do
+    GenServer.call(__MODULE__, {:new_node, node_name, port, sync_port}, 20000)
+  end
+
+  @spec sync_two_nodes(String.t(), String.t()) :: :ok
+  def sync_two_nodes(node_name1, node_name2) do
+    GenServer.call(__MODULE__, {:sync_two_nodes, node_name1, node_name2})
+    get_all_peers(node_name1)
+    get_all_peers(node_name2)
+  end
+
+  @doc """
+    Updates top block state for 2 nodes and compares them.
+    If top block of each node is equal - they are synced
+  """
+  @spec compare_nodes_by_top_block(String.t(), String.t()) :: :synced | :not_synced
+  def compare_nodes_by_top_block(node_name1, node_name2) do
+    get_node_top_block(node_name1)
+    get_node_top_block(node_name2)
+    GenServer.call(__MODULE__, {:compare_nodes_by_top_block, node_name1, node_name2})
+  end
+
+  @doc """
+    Updates registered oracles state for 2 nodes and compares them.
+    If registered oracles are equal - they are synced
+  """
+  def compare_nodes_by_registered_oracles(node_name1, node_name2) do
```

Comment:
I think we should probably do a compare for other parts of chainstate. Also maybe one function to compare whole chainstate would be enough. Also I'm not sure if this function is needed at all. I think if top block (or even top block hash) matches then it's guaranteed for chainstates to be equal as chainstate hash is part of block (of course this assumes #453 is fixed).